unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/NFTPromptHUD/NFTPromptHUDController.cs

```diff
@@ -1,43 +1,197 @@
+        view.OnOwnersTooltipFocusLost -= OnOwnersTooltipFocusLost;
+        view.OnOwnersTooltipFocus -= OnOwnersTooltipFocus;
+        view.OnViewAllPressed -= ShowOwnersPopup;
+        view.OnOwnersPopupClosed -= HideOwnersPopup;
+
+        CoroutineStarter.Stop(fetchNFTRoutine);
+
+        view?.Dispose();
 
         if (Environment.i != null)
             Environment.i.world.sceneController.OnOpenNFTDialogRequest -= OpenNftInfoDialog;
     }
+
+    private void SetNFT(NFTInfoSingleAsset info, string comment, bool shouldRefreshOwners)
+    {
+        lastNFTInfo = info;
+        view.SetNFTInfo(info, comment);
+        if (shouldRefreshOwners)
+        {
+            ownersInfoController.SetOwners(info.owners);
+        }
+    }
+
+    private void ShowOwnersTooltip()
+    {
+        isPointerInOwnerArea = true;
+
+        if (lastNFTInfo == null || lastNFTInfo.Value.owners.Length <= 1)
+            return;
+
+        var tooltip = view.GetOwnersTooltip();
+
+        if (tooltip.IsActive())
+        {
+            tooltip.KeepOnScreen();
+            return;
+        }
+
+        tooltip.SetElements(ownersInfoController.GetElements());
+        tooltip.Show();
+    }
+
+    void TryHideOwnersTooltip()
+    {
+        isPointerInOwnerArea = false;
+
+        if (!isPointerInTooltipArea)
+        {
+            HideOwnersTooltip();
+        }
+    }
+
+    void OnOwnersTooltipFocusLost()
+    {
+        isPointerInTooltipArea = false;
+        if (!isPointerInOwnerArea)
+        {
+            HideOwnersTooltip();
+        }
+    }
+
+    void OnOwnersTooltipFocus()
+    {
+        isPointerInTooltipArea = true;
+        var tooltip = view.GetOwnersTooltip();
+        if (tooltip.IsActive())
+        {
+            tooltip.KeepOnScreen();
+        }
+    }
+
+    private void HideOwnersTooltip() { HideOwnersTooltip(false); }
+
+    private void HideOwnersTooltip(bool instant)
+    {
+        var tooltip = view.GetOwnersTooltip();
+        if (!tooltip.IsActive())
+            return;
+
+        tooltip.Hide(instant);
+    }
+
+    private void ShowOwnersPopup()
+    {
+        HideOwnersTooltip(true);
+        isPointerInTooltipArea = false;
+        isPointerInOwnerArea = false;
+
+        var popup = view.GetOwnersPopup();
+
+        if (popup.IsActive())
+            return;
+
+        popup.SetElements(ownersInfoController.GetElements());
+        popup.Show();
+    }
+
+    private void HideOwnersPopup() { HideOwnersPopup(false); }
+
+    private void HideOwnersPopup(bool instant) { view.GetOwnersPopup().Hide(instant); }
+
+    internal static string FormatOwnerAddress(string address, int maxCharacters)
+    {
+        const string ellipsis = "...";
+
+        if (address.Length <= maxCharacters)
+        {
+            return address;
+        }
+        else
+        {
+            int segmentLength = Mathf.FloorToInt((maxCharacters - ellipsis.Length) * 0.5f);
+            return string.Format("{1}{0}{2}", ellipsis,
```

Comment:
Something really subjetive here, I prefer string interpolation, IMO it's more readable. You can give it a try to check if you like it as well.